sdk/eventhub/event-hubs/migrationguide.md

```diff
@@ -0,0 +1,215 @@
+    processInitialize: (initContext) => {
+      initContext.setStartingPosition(EventPosition.fromStart());
+    },
+    processEvents: onMessageHandler,
+    processError: onErrorHandler
+});
+  
+await subscription.close();
+```
+
+See [`receiveEvents.ts`](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) 
+for a sample program demonstrating this.
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient` for sending events
+
+In V2, there were multiple options on how to send data.
+
+In V5, this has been consolidated into a more efficient `sendBatch` method. 
+Batching merges information from multiple messages into a single send, reducing
+the amount of network communication needed vs sending messages one at a time.
+
+So in V2:
+```typescript
+const eventsToSend = [
+  // events go here
+];
+
+const client = EventHubClient.createFromConnectionString(connectionString);
+
+// Would fail if the total size of events exceed the max size supported by the library.
+await client.sendBatch(eventsToSend, partitionId);
+```
+
+In V5:
+```typescript
+const producer = new EventHubProducerClient(connectionString);
+
+const eventsToSend = [
+  // events go here
+];
+
+let batch = await producer.createBatch();
+let i = 0;
+
+while (i < eventsToSend.length) {
+  // messages can fail to be added to the batch if they exceed the maximum size configured for
+  // the EventHub.
+  const isAdded = batch.tryAdd(eventsToSend[i]);
+
+  if (isAdded) {
+    console.log(`Added event number ${i} to the batch`);
+    ++i;
+    continue;
+  }
+  
+  if (batch.count === 0) {
+    // If we can't add it and the batch is empty that means the message we're trying to send
+    // is too large, even when it would be the _only_ message in the batch.
+    //
+    // At this point you'll need to decide if you're okay with skipping this message entirely
+    // or find some way to shrink it.
+    console.log(`Message was too large and can't be sent until it's made smaller. Skipping...`);
+    ++i;
+    continue;
+  }
+
+  // otherwise this just signals a good spot to send our batch
+  console.log(`Batch is full - sending ${batch.count} messages as a single batch.`);
+  await producer.sendBatch(batch);
+
+  // and create a new one to house the next set of messages
+  batch = await producer.createBatch();
+}
+
+// send any remaining messages, if any.
+if (batch.count > 0) {
+  console.log(`Sending remaining ${batch.count} messages as a single batch.`)
+  await producer.sendBatch(batch);
+}
+```
+
+### Migrating code from `EventProcessorHost` to `EventHubConsumerClient` for receiving events
+
+In V2, `EventProcessorHost` allowed you to balance the load between multiple instances of your program when receiving events.
+
+In V5, `EventHubConsumerClient` allows you to do the same with `subscribe()` and the 
+`SubscriptionEventHandlers` interface in addition to passing a `CheckpointStore` into the constructor.
```

Comment:
I wonder if we should mention the `SubscriptionEventHandlers` interface here at all.
We don't end up referencing this interface in the sample, we just pass the callbacks.

How about

> In V5, `EventHubConsumerClient` allows you to do the same with the `subscribe()` method if you
pass a `CheckpointStore` to the constructor.